app/code/Magento/Customer/Model/Options.php

```diff
@@ -42,7 +43,10 @@ public function __construct(
      */
     public function getNamePrefixOptions($store = null)
     {
-        return $this->_prepareNamePrefixSuffixOptions($this->addressHelper->getConfig('prefix_options', $store));
+        return $this->prepareNamePrefixSuffixOptions(
+            $this->addressHelper->getConfig('prefix_options', $store),
+            $this->addressHelper->getConfig('prefix_show', $store) == NooptreqSource::VALUE_OPTIONAL
```

Comment:
Unfortunately we will need to change this since we cannot add the new constants.